tests/test_cli.py

```diff
@@ -501,15 +510,18 @@ def test_no_force_upload(mocker, testing_workdir, testing_metadata):
     api.output_yaml(testing_metadata, 'meta.yaml')
     args = ['--no-force-upload', testing_workdir]
     call = mocker.patch.object(conda_build.build.subprocess, 'call')
-    reset_context(testing_workdir)
-    main_build.execute(args)
-    pkg = api.get_output_file_path(testing_metadata)
-    assert call.called_once_with(['anaconda', 'upload', pkg])
-    args = [testing_workdir]
-    with open(os.path.join(testing_workdir, '.condarc'), 'w') as f:
-        f.write('anaconda_upload: True\n')
-    main_build.execute(args)
-    assert call.called_once_with(['anaconda', 'upload', '--force', pkg])
+    try:
+        _reset_config([os.path.join(testing_workdir, '.condarc')])
+        main_build.execute(args)
+        pkg = api.get_output_file_path(testing_metadata)
+        assert call.called_once_with(['anaconda', 'upload', pkg])
+        args = [testing_workdir]
+        with open(os.path.join(testing_workdir, '.condarc'), 'w') as f:
+            f.write('anaconda_upload: True\n')
+        main_build.execute(args)
+        assert call.called_once_with(['anaconda', 'upload', '--force', pkg])
+    finally:
+        _reset_config()
```

Comment:
Better to use pytest's [`request.addfinalizer`](https://docs.pytest.org/en/6.2.x/fixture.html#adding-finalizers-directly) for this via:

```python
request.addfinalizer(_reset_config)
```

Reply:
Right, thanks.
(And sorry for all those wacky edits; rarely a good idea to not run tests locally and doing other things concurrently...)